src/arrayMethodSort.js

```diff
@@ -4,8 +4,43 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function compareAsString(a, b) {
+    let x = a;
+    let y = b;
+
+    x = String(x);
+    y = String(y);
+
+    if (x > y) {
+      return 1;
+    }
+
+    if (x < y) {
+      return -1;
+    }
+
+    return 0;
+  }
+
+  [].__proto__.sort2 = function (compareFunction = compareAsString) {
```

Comment:
It's good practice to check if `compareFunction` is indeed a function before using it. You could use `typeof compareFunction === 'function'` to validate the input.